cmake/pcre.cmake

```diff
@@ -83,8 +83,17 @@ MACRO (CHECK_PCRE)
     IF(WITH_PCRE STREQUAL "system" OR WITH_PCRE STREQUAL "auto")
       FIND_PACKAGE(PkgConfig QUIET)
       PKG_CHECK_MODULES(PCRE libpcre2-8)
-      # in case pkg-config or libpcre2-8.pc is not installed:
-      CHECK_LIBRARY_EXISTS(pcre2-8 pcre2_match_8 "${PCRE_LIBRARY_DIRS}" HAVE_PCRE2_MATCH_8)
+      IF(NOT "${PCRE_LIBRARY_DIRS}" STREQUAL "")
+        SET(HAVE_PCRE2_MATCH_8 1)
+      ELSE()
+        # in case pkg-config or libpcre2-8.pc is not installed:
+        CHECK_LIBRARY_EXISTS(pcre2-8 pcre2_match_8 "" HAVE_PCRE2_MATCH_8)
+        IF(HAVE_PCRE2_MATCH_8)
+          FIND_LIBRARY(PCRE_LINK_LIBRARIES pcre2-8)
```

Comment:
I don't see how it could work. `-DCMAKE_PREFIX_PATH` only affects `FIND_LIBRARY()` but not `CHECK_LIBRARY_EXISTS()`, so the latter will fail if the the library is not in the standard linker path, and then `HAVE_PCRE2_MATCH_8` will be `NOT-FOUND` and `FIND_LIBRARY()` won't be used at all, whether `-DCMAKE_PREFIX_PATH` is set or not.

I'd simply remove `CHECK_LIBRARY_EXISTS()` and replace it with `FIND_LIBRARY()`